commands/debug.js

```diff
@@ -62,43 +62,65 @@ if(args[0].toLowerCase()==="quest"){
   message.channel.send("This creature can't do quests.");
   return;
 }
+
+var messagePing = message.mentions.members.first();
+var targetID = messagePing ? message.guild.id.concat(messagePing.id) : userid;
+charid = client.userMap.get(targetID,"possess");
+sburbid = client.charcall.allData(client,targetID,charid,"owner");
+
 if(args[0].toLowerCase()==="capgrist"){
-  var gristCheck = client.charcall.allData(client,userid,charid,"grist");
+  var gristCheck = client.charcall.allData(client,targetID,charid,"grist");
+  if(gristCheck == "NONE")
+  {
+    message.channel.send(`That person isn't currently possessing anything that has grist.`);
+    return;
+  }
+
+  if(sburbid == "NONE")
+  {
+    message.channel.send(`That one simply cannot be helped right now.`);
+    return;
+  }
+
   const gristTypes = ["build","uranium","amethyst","garnet","iron","marble","chalk","shale","cobalt","ruby","caulk","tar","amber","artifact","zillium","diamond"];
-  let rung = client.sburbMap.get(sburbid,"rung");
-  
   let max = 0;
-  if (client.sburbMap.get(sburbid,`godtier`)) {
-	if(args[1].toLowerCase()!="confirm")
-	{
-        message.channel.send(`I can only help godtiers by setting their invalid grist levels to 0. If you're sure, add a 'confirm' to that.`);
-		return;
-	}
+  if(sburbid && !client.sburbMap.get(sburbid,`godtier`))
+  {
+    max = client.cache(client.sburbMap.get(sburbid,"rung"));
   }
-  else {
-    max = client.cache(rung);
+  else if((!args[1] || args[1].toLowerCase()!="confirm") && (!args[2] || args[2].toLowerCase()!="confirm"))
+  {
+        message.channel.send(`I can only help that player by setting their invalid grist levels to 0. If you're sure, add a 'confirm' to that.`);
+        return;
   }
   
   for(i=0;i<gristTypes.length;i++){
 	if(isNaN(gristCheck[i])||isNull(gristCheck[i])||gristCheck[i]<0||(max > 0 && gristCheck[i]>max))
 		gristCheck[i] = max;
   }
-  client.charcall.setAnyData(client,userid,charid,gristCheck,"grist");
+  client.charcall.setAnyData(client,targetID,charid,gristCheck,"grist");
   message.channel.send("Fixed. Probably.");
   return;
 }
+
 if(args[0].toLowerCase()==="boons"){
   // Stupid JS syntax where value is set to 0 if undefined or something.
-  let currentBoons = client.charcall.allData(client,userid,charid,"b") || 0;
+  let currentBoons = client.charcall.allData(client,targetID,charid,"b") || 0;
```

Comment:
because of the way allData works, if it's not found this will return "NONE", which is a valid thing to store currentBoons as because typecasting isnt real in JS and life is terrible. So we need a check to see if currentBoons is NONE, then return with the fact the target can't hold boondollars if true.